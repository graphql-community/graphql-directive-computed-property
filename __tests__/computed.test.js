const { graphql } = require('graphql');
const nock = require('nock');
const directive = require('../index');
const schema = require('../example/schema');

beforeAll(() => {
  nock.disableNetConnect();
});

afterEach(() => {
  nock.cleanAll();
});

afterAll(() => {
  nock.enableNetConnect();
});

test('getDirectiveDeclaration should be defined', () => {
  expect(directive.getDirectiveDeclaration()).toMatchSnapshot();
});

test('if return computed property', async () => {
  const response = await graphql(
    schema,
    `
      query {
        me {
          firstName
          lastName
          fullName
        }
      }
    `
  );

  expect(response).toMatchSnapshot();
});

test('check if not replace original resolver', async () => {
  nock('https://yesno.wtf:443', { encodedQueryParams: true })
    .get('/api')
    .reply(200, { answer: 'yes' });

  const response = await graphql(
    schema,
    `
      query {
        me {
          firstName
          lastName
          admin
        }
      }
    `
  );

  expect(response).toMatchSnapshot();
});
