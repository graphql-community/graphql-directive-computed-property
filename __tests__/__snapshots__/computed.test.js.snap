// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if not replace original resolver 1`] = `
Object {
  "data": Object {
    "me": Object {
      "admin": "Are you admin? yes",
      "firstName": "John",
      "lastName": "Doe",
    },
  },
}
`;

exports[`getDirectiveDeclaration should be defined 1`] = `
GraphQLDirective {
  "args": Array [
    Object {
      "astNode": undefined,
      "defaultValue": undefined,
      "description": null,
      "name": "value",
      "type": "String",
    },
  ],
  "astNode": undefined,
  "description": undefined,
  "locations": Array [
    "FIELD_DEFINITION",
  ],
  "name": "rest",
}
`;

exports[`if return computed property 1`] = `
Object {
  "data": Object {
    "me": Object {
      "firstName": "John",
      "fullName": "John Doe",
      "lastName": "Doe",
    },
  },
}
`;
